File: 엄마호랑이반/8주차 문제/장원석/삼각달팽이.java
import java.util.Arrays;

public class 삼각달팽이 {

	public static void main(String[] args) {
		System.out.println(Arrays.toString(solution(4)));
		System.out.println(Arrays.toString(solution(5)));
		System.out.println(Arrays.toString(solution(6)));

	}

	static int[] dx = { 1, 0, -1 };
	static int[] dy = { 0, 1, -1 };

	private static int[] solution(int n) {

		int count = n * (n + 1) / 2;
		int[][] map = new int[n][n];

		int x = -1, y = 0;  // 시작 위치
	    int value = 1;
	    int direction = 0;
	    
	    for(int moveLen = n; moveLen > 0; moveLen--) {
	        for(int i = 0; i < moveLen; i++) {
	            x += dx[direction];
	            y += dy[direction];
	            map[x][y] = value++;
	        }
	        
	        direction = (direction + 1) % 3;
	    }

		// 1차원 배열로 변환
		int[] answer = new int[count];
		int index = 0;

		for (int i = 0; i < n; i++) {
			for (int j = 0; j <= i; j++) {
				answer[index++] = map[i][j];
			}
		}

		return answer;
	}
}

File: 엄마호랑이반/8주차 문제/박정환/삼각달팽이.py
def solution(n):
    answer = []
    a = [[0] * n for i in range(n)]
    x , y = -1, 0
    num = 1

    for i in range(n):
        # 0, 1, 2, 3
        for _ in range(i, n):
            if i % 3 == 0:
                x += 1
            elif i % 3 == 1:
                y += 1
            else:
                x -= 1
                y -= 1
            a[x][y] = num
            num+=1

    for i in range(n):
        for j in range(i+1):
            answer.append(a[i][j])
    return answer
